math/fastExponent.go

```diff
@@ -1,31 +1,19 @@
 package main
 
-//based on square and multiply algorithm
-
-func fastExponentiation(b int, e int) float64 {
-	//runs in O(log(n)) where n = e
-	if e == 0 {
-		return 1.0
-	}
-	if e == 1 {
-		return float64(b)
-	}
-	if e < 0 {
-		e *= -1
-		println("executed")
-		return 1 / fastExponentiation(b, e)
-	}
-	r := 1
-	for e > 0 {
-		if e%2 == 1 {
-			r = r * b
+func fastExponentiation(x int, y int) float64 {
+	res := 1 // res is the final result
```

Comment:
Remove unnecessary comments. Please add more references to the implementation (describe how it work, or refer to other document explain how it work. Naming things meaningful also. 

Could you remove the main function and write some testcases for this implementation?